unlock-app/src/paywall-builder/build.js

```diff
@@ -9,9 +20,30 @@ export default function buildPaywall(window, document, lockAddress, blocker) {
 
   const paywallUrl = findPaywallUrl(document) + `/paywall/${lockAddress}/`
   const iframe = getIframe(document, paywallUrl)
-
   add(document, iframe)
 
+  // iOS allows the page to scroll even when the paywall is up. Our
+  // solution to this is to make the paywall grow to obscure the page
+  // content as the user scrolls down the page. We register the
+  // scroll-handling function here to preserve the context of the actual
+  // page, not the iframe.
+  const scrollLoop = () => {
+    const pageTop = window.pageYOffset
+    const viewportHeight = window.innerHeight
+    const pageHeight = document.documentElement.scrollHeight
+    const maximumScroll = pageHeight - viewportHeight
+    // Avoiding a divide-by-zero error when the page does not scroll!
+    if (maximumScroll === 0) {
+      return
+    }
+
+    const scrollPosition = baseBannerHeight() + 100 * (pageTop / maximumScroll)
+    iframe.contentWindow.postMessage({ scrollPosition }, '*')
```

Comment:
just as a quick note, using `'*'` is a potential security hole, but I am going to handle this in a separate PR that combines all the `postMessage` stuff into a middleware, so this is the best solution for this PR.